client/client-lib/src/lib.rs

```diff
@@ -345,21 +345,45 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
         address
     }
 
-    /// **WARNING** this selects and removes coins from the database without confirming whether
-    /// we have successfully spent them in a transaction.
-    pub fn select_and_spend_coins(&self, amount: Amount) -> Result<TieredMulti<SpendableNote>> {
-        let mut batch = DbBatch::new();
-        let mut tx = batch.transaction();
+    /// Issues a spendable amount of ecash
+    ///
+    /// **WARNING** the ecash will be deleted from the database, the returned ecash must be
+    /// `reissued` or it will be lost
+    pub async fn spend_ecash<R: RngCore + CryptoRng>(
+        &self,
+        amount: Amount,
+        mut rng: R,
+    ) -> Result<TieredMulti<SpendableNote>> {
         let coins = self.mint_client().select_coins(amount)?;
-        tx.append_from_iter(coins.iter_items().map(|(amount, coin)| {
-            BatchItem::delete(CoinKey {
+        let mut tx = TransactionBuilder::default();
+        tx.input_coins(coins, &self.context.secp)?;
+        tx.output_coins(
+            amount,
+            &self.mint_client().context.secp,
+            &self.mint_client().config.tbs_pks,
+            &mut rng,
+        );
+        self.submit_tx_with_change(tx, DbBatch::new(), rng).await?;
```

Comment:
Ideally we could attempt to `amount` worth of ecash and only submit a transaction to consensus if that fails. But I'm not sure if there's an API for that right now.

Reply:
Just added that shortcut if the client already has exact change.